syntax = "proto3";

package syft.core.node.common.service;

import "proto/core/common/common_object.proto";
import "proto/core/io/address.proto";
import "proto/core/pointer/pointer.proto";

message ObjectSearchMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  syft.core.io.Address reply_to = 3;
}

message ObjectSearchReplyMessage {
  syft.core.common.UID msg_id = 1;
  syft.core.io.Address address = 2;
  repeated syft.core.pointer.Pointer results = 3;
}
